src/digests.rs

```diff
@@ -282,6 +282,26 @@ pub trait MultihashDigest {
     ///
     /// Panics if the digest length is bigger than 2^32. This only happens for identity hasing.
     fn digest(&self, data: &[u8]) -> Multihash;
```

Comment:
One thing I like about the original `Digest` trait is that `digest` actually does not take `self`, but rather only the data, such that you can write `Sha2_256::digest([1u8; 14])` without going through creating an instance. I wonder if we should do this switch as well, to make the pi more ergonomic.

Reply:
Isn't that still the case, with this code: https://github.com/multiformats/rust-multihash/pull/58/files#diff-5b11d6021f0af96d8da7e3de8ca657b4 Line 77-80 (sorry I can't directly link due to GitHub not showing the diff by default.

Reply:
I re-read your comment, I think I now got the point. The reason to use `self` is, that it can be used as a trait object. I see the `MutlihashDigest` main use as a trait object, else you would use the impl directly, without the trait.

Reply:
I guess, but then I think having the `fn digest(&self)` version makes even less sense

Reply:
So `MultihashDigest` is kind of what https://docs.rs/digest/0.8.1/digest/trait.DynDigest.html is.

Reply:
The idea is that you can call `digest()` based on a multihash (or just the code of it) without the need to know which algorithm it actually is: `my_multihash.algorithm().hasher()?.digest("something")`.

Reply:
I see, that works for me, but should probably be documented somewhere :)